rtwcli/rtw_cmds/rtw_cmds/workspace/create_verb.py

```diff
@@ -0,0 +1,828 @@
+# Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+from dataclasses import dataclass, fields
+import os
+import pathlib
+from pprint import pprint
+import shutil
+import textwrap
+from typing import Any, List
+import questionary
+from rtwcli.constants import (
+    BASHRC_PATH,
+    ROS_TEAM_WS_GIT_HTTPS_URL,
+    SKEL_BASHRC_PATH,
+    WORKSPACES_PATH,
+)
+from rtwcli.docker_utils import (
+    change_docker_path_permissions,
+    docker_build,
+    docker_cp,
+    docker_exec_bash_cmd,
+    docker_stop,
+    is_docker_tag_valid,
+)
+from rtwcli.utils import (
+    create_file_and_write,
+    create_temp_file,
+    git_clone,
+    run_bash_command,
+    run_command,
+    vcs_import,
+)
+from rtwcli.verb import VerbExtension
+import docker
+from rtwcli.workspace_manger import (
+    Workspace,
+    WorkspacesConfig,
+    get_compile_cmd,
+    get_workspace_names,
+    save_workspaces_config,
+    update_workspaces_config,
+)
+
+
+DEFAULT_BASE_IMAGE_NAME_FORMAT = "osrf/ros:{ros_distro}-desktop"
+DEFAULT_INTERMEDIATE_IMAGE_NAME_FORMAT = "rtw_{base_image_name}_intermediate"
+DEFAULT_FINAL_IMAGE_NAME_FORMAT = "rtw_{workspace_name}_final"
+DEFAULT_CONTAINER_NAME_FORMAT = "{final_image_name}-instance"
+DEFAULT_RTW_DOCKER_BRANCH = "rtw_ws_create"
+DEFAULT_RTW_DOCKER_PATH = os.path.expanduser("~/ros_team_workspace")
+DEFAULT_UPSTREAM_WS_NAME_FORMAT = "{workspace_name}_upstream"
+DEFAULT_WS_REPOS_FILE_FORMAT = "{repo_name}.{ros_distro}.repos"
+DEFAULT_UPSTREAM_WS_REPOS_FILE_FORMAT = "{repo_name}.{ros_distro}.upstream.repos"
+DEFAULT_APT_PACKAGES = [
+    "bash-completion",
+    "git",
+    "git-lfs",
+    "iputils-ping",
+    "nano",
+    "python3-colcon-common-extensions",
+    "python3-pip",
+    "python3-vcstool",
+    "sudo",
+    "tmux",
+    "trash-cli",
+    "tree",
+    "vim",
+    "wget",
+]
+DEFAULT_PYTHON_PACKAGES = ["pre-commit"]
+DEFAULT_SSH_ABS_PATH = os.path.expanduser("~/.ssh")
+DEFAULT_INTERMEDIATE_DOCKERFILE_NAME = "Dockerfile"
+DEFAULT_INTERMEDIATE_DOCKERFILE_SAVE_FOLDER_FORMAT = "{ws_folder}/docker"
+
+
+@dataclass
+class CreateVerbArgs:
+    ws_abs_path: str
+    ros_distro: str
+    docker: bool = False
+    repos_location_url: str = None
```

Comment:
Only comment I have for now is to change this variable to be clearer, we have to enter here the base repository URL where the repos file is placed, and not full url to the `.repos` file.